medocker/docker_shared/run_warpx.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+
+# This script handles running MPI/Warp in docker containers on AWS, and the
+# associated copy and monitoring operations that accompany it. It detects runs
+# that don't output to stdout for $TIMEOUT seconds and will terminate at that
+# point. It will move files either as the run is ongoing, or just at the end,
+# back to S3. Dump files are never moved or copied to S3; they will be copied
+# from S3 if they're already there.
+
+# S3 can sometimes make directories with no name ("") if two slashes are
+# present in sequence
+# https://stackoverflow.com/questions/1848415/remove-slash-from-the-end-of-a-variable
+DIRNAME=${1%/}
+BUCKET=$2
+SCRIPTNAME='run_script.sh'
+
+# MVFILES can be an environment variable set to True to move files as
+# generated.
+echo "Move diagnostic files as generated:" $MVFILES
+
+# TIMEOUT can be an environment variable. If no output to stdout has been made
+# in the previous TIMEOUT seconds, this script terminates.
+# TIMEOUT defaults to 3h = 10800 s
+# https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash
+if [ -z ${TIMEOUT:x} ] || ! (( "$TIMEOUT" > 0 )); then
+    echo TIMEOUT set to default of 10800 sec
+    TIMEOUT=10800
+else
+    echo TIMEOUT set to "$TIMEOUT"
+fi
+
+# Use EFS directory as working directory.
+echo Make directory ${DIRNAME} in EFS and cd there
+mkdir -p /efs/WarpX_simulation_runs/${DIRNAME}
+cd /efs/WarpX_simulation_runs/${DIRNAME}
+
+# Copy files from S3
+aws s3 cp --recursive s3://${BUCKET}/${DIRNAME} ./ --exclude "*" \
+    --include "run*" --include "*.dump" --include "std*" \
+    --include "*runinfo.dpkl"
+ls
+
+# Move diagnostic files that are done
+function checkptmove {
+    # Check file is not open. Since they shouldn't be re-opened, this should be
+    # fine.
+    if ! lsof $1 > /dev/null
+    then
+        # http://www.linuxjournal.com/content/normalizing-path-names-bash
+        # Remove all /./ and // sequences.
+        local path=${DIRNAME}/"$1"
+        path=${path//\/.\//\/}
+        path=${path//\/\//\/}
+        path=${path/diags_[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]/diags}
+
+        # Remove dir/.. sequences.
+        while [[ $path =~ ([^/][^/]*/\.\./) ]]
+        do
+            path=${path/${BASH_REMATCH[0]}/}
+        done
+        echo $path
+        aws s3 mv $1 s3://${BUCKET}/${path}
+    fi
+}
+
+# Every 5 minutes, move (to keep space free, if $MVFILES is True) or sync
+# per-timestep diagnostic files as they are done being written. Always sync
+# other files (which may still be opened!)
+function checkpoint {
+while true
+do
+    sleep 300
+    if [[ "$MVFILES" == "True" ]]
+    then
+        find . -name "*Header" | while read file; do checkptmove "$file"; done
+        find . -name "Cell_D_*[0-9][0-9][0-9][0-9][0-9]" | while read file; do checkptmove "$file"; done
+        find . -name "warpx_job_info" | while read file; do checkptmove "$file"; done
```

Comment:
We should have a discussion about whether we think copying the full diagnostic outputs to S3 is what we want to do since it will result in a lot of storage usage. Perhaps we only need to copy summarized data.

Reply:
Definitely, since we're using elastic storage now. I believe that this script should still be fine as is, because it allows users to choose either to copy the diagnostic files over or move them. To prevent redundant use of space, we could use the move functionality instead of copy.